source/neuropod/bindings/java/src/main/native/com_uber_neuropod_NeuropodTensor.cc

```diff
@@ -109,3 +112,90 @@ JNIEXPORT jlong JNICALL Java_com_uber_neuropod_NeuropodTensor_nativeGetNumberOfE
+        auto tensorType = neuropodTensor->get_tensor_type();
+        if (tensorType != neuropod::STRING_TENSOR)
+        {
+            throw std::runtime_error("unexpected tensor type, should be STRING_TENSOR but found:" +
+                                     tensorTypeToString(tensorType));
+        }
+
+        auto    size = neuropodTensor->get_num_elements();
+        jobject ret  = env->NewObject(java_util_ArrayList, java_util_ArrayList_, size);
+        if (!ret)
+        {
+            throw std::runtime_error("out of memory: cannot create ArrayList");
+        }
+
+        auto typedTensor = neuropodTensor->as_typed_tensor<std::string>();
+
+        std::function<void(string_accessor_type *)> mapFunc = [env, ret](string_accessor_type *elem) {
+            // StringProxy supports conversion to std:string.
+            std::string tmpStr(*elem);
+            jstring     convertedElem = env->NewStringUTF(tmpStr.c_str());
+            env->CallBooleanMethod(ret, java_util_ArrayList_add, convertedElem);
+            env->DeleteLocalRef(convertedElem);
+        };
+
+        auto dims = typedTensor->get_dims();
+        switch (dims.size())
+        {
+        case 1:
+            mapStringTensor(typedTensor->accessor<1>(), mapFunc, dims);
+            break;
+        case 2:
+            mapStringTensor(typedTensor->accessor<2>(), mapFunc, dims);
+            break;
+        case 3:
+            mapStringTensor(typedTensor->accessor<3>(), mapFunc, dims);
+            break;
+        case 4:
+            mapStringTensor(typedTensor->accessor<4>(), mapFunc, dims);
+            break;
+        default:
+            // Here copy data twice
+            const auto &elementList = typedTensor->get_data_as_vector();
+            for (const auto &elem : elementList)
+            {
+                jstring convertedElem = env->NewStringUTF(elem.c_str());
+                env->CallBooleanMethod(ret, java_util_ArrayList_add, convertedElem);
+                env->DeleteLocalRef(convertedElem);
+            }
+        }
+        return ret;
+    }
+    catch (const std::exception &e)
+    {
+        throwJavaException(env, e.what());
+    }
+    return nullptr;
+}
+
+JNIEXPORT jstring JNICALL Java_com_uber_neuropod_NeuropodTensor_nativeGetString(JNIEnv *env,
+                                                                                jclass,
+                                                                                jlong index,
+                                                                                jlong handle)
+{
+    try
+    {
+        auto stringTensor = (*reinterpret_cast<std::shared_ptr<neuropod::NeuropodValue> *>(handle))
+                                ->as_tensor()
+                                ->as_typed_tensor<std::string>();
+        const auto &       strList = stringTensor->get_data_as_vector();
+        const std::string &elem    = strList[index];
```

Comment:
This is super inefficient if we just need one element (because it makes a copy of the entire tensor). Similar to above, let's just add this as a friend function of `TypedNeuropodTensor<std::string>` so we can just get the single element

Reply:
Added get_data_at(index). Let me know what you think about this approach.